src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java

```diff
@@ -53,4 +55,52 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
         // convert internal representation of user back to API
         return DTOMapper.INSTANCE.convertEntityToUserGetDTO(createdUser);
     }
+
+    @GetMapping("/user/{userId}")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public UserGetDTO getUserByUserName(@RequestHeader("X-Auth-Token") String token, @PathVariable Long userId){
+        UserGetDTO userGetDTO = new UserGetDTO();
+        return  userGetDTO;
+
+    }
+
+    @GetMapping("/user/login")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @ResponseBody
+    public UserAuthDTO login(@RequestBody UserPostDTO userPostDTO){
+        UserAuthDTO userAuthDTO = new UserAuthDTO("supersecrettokenvalue");
+        return userAuthDTO;
+    }
+
+    @PutMapping("/logout/{userId}")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void logout(@RequestHeader("X-Auth-Token") String token,@PathVariable Long userId){
+        return;
+    }
+
+    @PutMapping("/user/{userId}/edit")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public UserUpdateDTO updateUser(@RequestHeader("X-Auth-Token") String token, @RequestBody User user, @PathVariable Long userId){
```

Comment:
The RequestBody should be a DTO object, just create one if it does not fit the class diagram scheme for now, we can update it later again.

Reply:
Changed to UpdateUserDTO